### ISSUE TITLE
Add a way to specify the swift-tools version.

### ISSUE THREAD

**Issue by catshow** [CreateAPI/CreateAPI#162]
We are using swift-tools-version:5.7, and CreateAPI keeps overriding this information in the Packages.swift file.

Please add a section to the config specification for:
swift:
  tools-version: 5.7

**Comment by liamnichols** [comment]
Thanks for raising this @catshow! Just to confirm, what is the impact of the Swift Tools Version being set back to 5.6? 

It shouldn't prevent the package from being used does it? 

If there is a valid use case for it though, it would make a nice configuration option 👍

**Comment by catshow** [comment]
Yes it is resetting things back to 5.5 actually.  I am using the 1.1 binary of CreateAPI, using a package plugin for my builds.

**Comment by liamnichols** [comment]
So you want to add the plugin target to the generated Package.swift? 

The intention of using CreateAPI to generate a complete package is that you then use that package as a dependency elsewhere in other packages rather than modifying it directly. Because otherwise if you make modifications then they’ll be overwritten when you run the generator the next time.

Would adding the plugin to your own separate package address this instead? Or am I misunderstanding the use case?

**Comment by LePips** [comment]
As far as I understand, they just want to change the swift package tool version in the `Package.swift` definition, which is currently hardcoded to 5.5 in our templates.

**Comment by liamnichols** [comment]
Right I see. It’s my understanding however that the tools version is just an indicator of the minimum supported version?

So having it set to a lower value shouldn’t block anything as long as we aren’t trying to use new features (such as package plugins) in the Package itself? 

CreateAPI generates a package that’s compatible with 5.5 and later so I’m not sure why it would need changing assuming that you’re not modifying the generated package manually? 

I might have misunderstood things though, that’s why I’m trying to clarify
